src/csharp/NativeMethods.cs

```diff
@@ -0,0 +1,90 @@
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaGeneratorParamsSetInputIDs(IntPtr /* OgaGeneratorParams* */ generatorParams,
+                                                                                   int[] /* int32_t* */ inputIDs,
+                                                                                   UIntPtr inputIDsCount,
+                                                                                   UIntPtr sequenceLength,
+                                                                                   UIntPtr batchSize);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaCreateGenerator(IntPtr /* OgaModel* */ model,
+                                                                        IntPtr /* OgaGeneratorParams* */ generatorParams,
+                                                                        out IntPtr /* OgaGenerator** */ generator);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern void OgaDestroyGenerator(IntPtr /* OgaGenerator* */ generator);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern bool OgaGenerator_IsDone(IntPtr /* OgaGenerator* */ generator);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaGenerator_ComputeLogits(IntPtr /* OgaGenerator* */ generator);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaGenerator_GenerateNextToken_Top(IntPtr /* OgaGenerator* */ generator);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern IntPtr /* OgaResult* */ OgaGenerator_GetSequence(IntPtr /* OgaGenerator* */ generator,
+                                                                              int index,
+                                                                              IntPtr /* int32_t* */ tokens,
+                                                                              out UIntPtr /* size_t* */ tokensCount);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern void OgaDestroySequences(IntPtr /* OgaSequences* */ sequences);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern ulong OgaSequencesCount(IntPtr /* OgaSequences* */ sequences);
+
+        [DllImport(NativeLib.DllName, CallingConvention = CallingConvention.Winapi)]
+        public static extern ulong OgaSequencesGetSequenceCount(IntPtr /* OgaSequences* */ sequences,
```

Comment:
This API must return UIntPtr and accept UIntPtr.

Reply:
Addressed this.